pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -499,6 +500,15 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "the enable-ds-eviction annotation cannot be deleted",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				if daemonset.Spec.Template.Annotations != nil {
+					delete(daemonset.Spec.Template.Annotations, "cluster-autoscaler.kubernetes.io/enable-ds-eviction")
+				}
+			},
+			expect: false,
```

Comment:
The test input doesn't have the annotation set, so this test passes because `delete` doesn't actually modify the input.  But shouldn't `daemonsetConfigChanged` return `true` if `cluster-autoscaler.kubernetes.io/enable-ds-eviction` is deleted, so that the operator adds the missing annotation?  

What this test needs to do is verify that `daemonsetconfigChanged` detects changes to annotations that the operator manages, and that it ignores changes to annotations that the operator doesn't manage.  The operator manages two annotations: `target.workload.openshift.io/management` and `cluster-autoscaler.kubernetes.io/enable-ds-eviction`.  What we could do is modify the test input so it has `target.workload.openshift.io/management` set and then modify this new test case to delete the annotation.  Then the new test case would verify that the operator detects the case (a) that an operator-managed annotation has been _deleted_, and already the preceding test case verifies that the operator detects the case (b) that an operator-managed annotation has been _added_, and the test case before that verifies that the operator ignores the case that (c) an unmanaged annotation is set.

Reply:
I neglected to notice that this test didn't set it initially, so I can change that.  The `target.workload.openshift.io/management annotation` is not in the `managedDNSDaemonSetAnnotations`, so it does not register a config change in `daemonsetConfigChanged`.  Is that another bug?

Reply:
Also, someone should not be able to change the value of `luster-autoscaler.kubernetes.io/enable-ds-eviction` to false nor to delete it, right?

Reply:
@Miciah I made some changes to the code and unit tests.  PTAL.

Reply:
> I neglected to notice that this test didn't set it initially, so I can change that. The `target.workload.openshift.io/management annotation` is not in the `managedDNSDaemonSetAnnotations`, so it does not register a config change in `daemonsetConfigChanged`. Is that another bug?

Yeah, that's another bug.  

> Also, someone should not be able to change the value of `cluster-autoscaler.kubernetes.io/enable-ds-eviction` to false nor to delete it, right?

Correct.